services/121-service/src/registration/registrations.service.ts

```diff
@@ -882,37 +887,131 @@ export class RegistrationsService {
+        FinancialServiceProviderName.intersolveVisa,
+      );
+    const intersolveVisaQuestionNames = intersolveVisaQuestions.map(
+      (q) => q.name,
+    );
+    const dataFieldNames = [
+      'fullName',
+      'phoneNumber',
+      ...intersolveVisaQuestionNames,
+    ];
+
+    const registrationData =
+      await this.registrationDataScopedRepository.getRegistrationDataArrayByName(
+        registration,
+        dataFieldNames,
+      );
+
+    if (!registrationData || registrationData.length === 0) {
+      throw new HttpException(
+        `No registration data found for referenceId: ${referenceId}`,
+        HttpStatus.NOT_FOUND,
+      );
+    }
+
+    const mappedRegistrationData = registrationData.reduce(
+      (acc, { name, value }) => {
+        acc[name] = value;
+        return acc;
+      },
+      {},
+    );
+
+    // // TODO: Test with debugger to see empty properties
+    // // Check if all required properties are present. If not, create a failed transaction and throw an error.
+    for (const key in registrationData) {
+      if (key === 'addressHouseNumberAddition') continue; // Skip non-required property
+
+      // Define "empty" based on your needs. Here, we check for null, undefined, or an empty string.
+      if (
+        registrationData[key] === null ||
+        registrationData[key] === undefined ||
+        // TODO: Is this .value check what we want?
+        registrationData[key].value === ''
+      ) {
+        const errorText = `Property ${key} is undefined`;
+        throw new HttpException(errorText, HttpStatus.BAD_REQUEST);
+      }
+    }
+    const reissueCardDto: ReissueCardDto = {
+      registrationId: registration.id,
+      // Why do we need this?
+      reference: registration.referenceId,
+      personalData: {
+        name: mappedRegistrationData['fullName'],
+        addressStreet: mappedRegistrationData['addressStreet'],
+        addressHouseNumber: mappedRegistrationData['addressHouseNumber'],
+        addressHouseNumberAddition:
+          mappedRegistrationData['addressHouseNumberAddition'],
+        addressPostalCode: mappedRegistrationData['addressPostalCode'],
+        addressCity: mappedRegistrationData['addressCity'],
+        phoneNumber: mappedRegistrationData['phoneNumber'], // In the above for loop it is checked that this is not undefined or empty
+      },
+      brandCode: intersolveVisaConfig.find(
+        (c) => c.name === FinancialServiceProviderConfigurationEnum.brandCode,
+      )?.value as string, // This must be a string. If it is not, the intersolve API will return an error (maybe).
+      coverLetterCode: intersolveVisaConfig.find(
+        (c) =>
+          c.name === FinancialServiceProviderConfigurationEnum.coverLetterCode,
+      )?.value as string, // This must be a string. If it is not, the intersolve API will return an error (maybe).
+    };
+
+    await this.intersolveVisaService.reissueCard(reissueCardDto);
```

Comment:
Instead of creating the DTO object first and then inputting that into the method, in other places I create the object in the method call. According to Dom better, as this then does not need the DTO import and is better readable. I do not necessarily agree, but do think 1 way to do this everywhere is better for readability than various ways throughout the code.